.github/workflows/format.yml

```diff
@@ -11,6 +11,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: DoozyX/clang-format-lint-action@v0.16.2
       with:
+        extensions: 'h,cpp,c,td'
```

Comment:
I'm not sure we want to do this for the .td files. Thoughts?

Reply:
Thats the whole point for the PR.

Reply:
I guess thats about taste. And having one fixed representation of the code.

Reply:
Yeah I can see pros and cons, and just haven't thought about it deeply enough that I'm curious for your thoughts.

cc @ftynse @umatin @pengmai re MLIR since that will be substantial amts of tablegen.

Reply:
I must admit though that some of the formatted code blocks look quite shabby.